core/src/main/scala/pureconfig/PeriodUtils.scala

```diff
@@ -0,0 +1,61 @@
+package pureconfig
+
+import java.time.Period
+import java.time.temporal.ChronoUnit
+import java.time.temporal.ChronoUnit._
+
+import scala.util.Try
+
+import com.typesafe.config.impl.ConfigImplUtil
+import pureconfig.error.{ CannotConvert, FailureReason }
+
+/**
+ * Utility functions for converting a `String` to a `Period`. The parser accepts the HOCON unit syntax.
+ */
+private[pureconfig] object PeriodUtils {
+
+  /**
+   * Convert a string to a Period while trying to maintain compatibility with Typesafe's abbreviations.
+   */
+  val fromString: String => Either[FailureReason, Period] = { str =>
+    Try(Right(Period.parse(str))).getOrElse(typesafeConfigParsePeriod(str))
+  }
+
+  // This is a copy of the period parser in `com.typesafe.config.impl.SimpleConfig` adapted to be safer (dealing with
+  // `Either` values instead of throwing exceptions). Try not to refactor this too much so as to be easy for anyone
+  // to compare this code with the original.
+  def typesafeConfigParsePeriod(input: String): Either[FailureReason, Period] = {
+    val s = ConfigImplUtil.unicodeTrim(input)
+    val originalUnitString = getUnits(s)
+    var unitString = originalUnitString
+    val numberString = ConfigImplUtil.unicodeTrim(s.substring(0, s.length - unitString.length))
+    var units: ChronoUnit = null
+    // this would be caught later anyway, but the error message
+    // is more helpful if we check it here.
+    if (numberString.length == 0) return Left(CannotConvert(input, "Period", "No number in period value"))
+    if (unitString.length > 2 && !unitString.endsWith("s")) unitString = unitString + "s"
+    // note that this is deliberately case-sensitive
+    if (unitString == "" || unitString == "d" || unitString == "days") units = ChronoUnit.DAYS
```

Comment:
Translating an empty string to days seems weirdly dangerous. I know you're just copying what typesafe config did, but surprises me that they made that choice. Do you know what the rationale was for defaulting to days?

Reply:
Not sure, but it's [well documented](https://github.com/lightbend/config/blob/master/HOCON.md#period-format), so I don't think we should decide against it. If I was to guess, defaulting to days results in HOCON also accepting integers in `Period` configurations, a type that's arguably nicer to handle when generating or modifying configs programatically.